src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -127,37 +126,22 @@ private async Task<BundleResponse> ExecuteTransactionForAllRequests(Hl7.Fhir.Mod
             return new BundleResponse(responseBundle.ToResourceElement());
         }
 
-        private static void ThrowTransactionException(HttpContext httpContext, OperationOutcome operationOutcome)
+        private async Task FillRequestLists(Hl7.Fhir.Model.Bundle.BundleType? bundleType, List<Hl7.Fhir.Model.Bundle.EntryComponent> bundleEntries)
         {
-            var operationOutcomeIssues = GetOperationOutcomeIssues(operationOutcome.Issue);
+            var resourceUrlList = new HashSet<string>();
 
-            var errorMessage = string.Format(Api.Resources.TransactionFailed, httpContext.Request.Method, httpContext.Request.Path);
-
-            throw new TransactionFailedException(errorMessage, (HttpStatusCode)httpContext.Response.StatusCode, operationOutcomeIssues);
-        }
-
-        private static List<OperationOutcomeIssue> GetOperationOutcomeIssues(List<OperationOutcome.IssueComponent> operationoutcomeIssueList)
-        {
-            var issues = new List<OperationOutcomeIssue>();
-
-            operationoutcomeIssueList.ForEach(x =>
-                issues.Add(new OperationOutcomeIssue(
-                    x.Severity.ToString(),
-                    x.Code.ToString(),
-                    x.Diagnostics)));
-
-            return issues;
-        }
-
-        private async Task FillRequestLists(List<Hl7.Fhir.Model.Bundle.EntryComponent> bundleEntries)
-        {
             foreach (Hl7.Fhir.Model.Bundle.EntryComponent entry in bundleEntries)
             {
                 if (entry.Request.Method == null)
                 {
                     continue;
                 }
 
+                if (bundleType == Hl7.Fhir.Model.Bundle.BundleType.Transaction)
```

Comment:
[Spec](http://hl7.org/fhir/http.html#brules) says:

> For a batch, there SHALL be no interdependencies between the different entries in the Bundle that cause change on the server. The success or failure of one change SHOULD not alter the success or failure or resulting content of another change.

But seems like we are implemeting this for transaciton?

Reply:
In the [transaction section](http://hl7.org/fhir/http.html#trules) it states:

> If any resource identities (including resolved identities from conditional update/delete) overlap in steps 1-3, then the transaction SHALL fail.

Steps 1-3 refers to 

> 1. Process any DELETE interactions
> 2. Process any POST interactions
> 3. Process any PUT or PATCH interactions

Reply:
Okay so for this PR, it's only addressing the transaction part and there is another work item to track for batch?

Reply:
I checked the behavior on other fhir test servers but didnot get the answer i am looking for. For now, i will create a separate story for batch.